three/three.cpp

```diff
@@ -400,42 +449,72 @@ std::string Three::_fetchPythonError() const {
     return ret_val;
 }
 
-char *Three::_getCheckVersion(PyObject *module) const {
-    if (module == NULL) {
+char *Three::_getStringAttr(PyObject *obj, const char *attributeName) const {
+    if (obj == NULL) {
         return NULL;
     }
 
     char *ret = NULL;
-    PyObject *py_version = NULL;
-    PyObject *py_version_bytes = NULL;
-    char version_field[] = "__version__";
-
-    // try getting module.__version__
-    py_version = PyObject_GetAttrString(module, version_field);
-    if (py_version != NULL && PyUnicode_Check(py_version)) {
-        py_version_bytes = PyUnicode_AsEncodedString(py_version, "UTF-8", "strict");
-        if (py_version_bytes == NULL) {
-            setError("error converting __version__ to string: " + _fetchPythonError());
+    PyObject *py_attr = NULL;
+    PyObject *py_attr_bytes = NULL;
+
+    py_attr = PyObject_GetAttrString(obj, attributeName);
+    if (py_attr != NULL && PyUnicode_Check(py_attr)) {
+        py_attr_bytes = PyUnicode_AsEncodedString(py_attr, "UTF-8", "strict");
+        if (py_attr_bytes == NULL) {
+            setError("error converting " + std::string(attributeName) + " to string: " + _fetchPythonError());
             ret = NULL;
             goto done;
         }
-        ret = _strdup(PyBytes_AsString(py_version_bytes));
+        ret = _strdup(PyBytes_AsString(py_attr_bytes));
         goto done;
     } else {
-        // we expect __version__ might not be there, don't clutter the error stream
+        // we expect that attribute might not be there, don't clutter the error stream
+        // TODO: detect the type of __version__ and set the correct error
```

Comment:
I don't think we should care why the version was wrong, let's remove these lines